geoutils/geovector.py

```diff
@@ -338,15 +339,99 @@ def query(self: VectorType, expression: str, inplace: bool = False) -> VectorTyp
+
+        By default, the boundary of the Vector's geometry will be used. The full geometry can be used by passing "geometry",
+        or any lower dimensional geometry attribute such as "centroid", "envelope" or "convex_hull".
+        See all geometry attributes in the Shapely documentation at https://shapely.readthedocs.io/.
+
+        :param raster: Raster to burn the proximity grid on, defaults to a 1000 x 1000 pixel grid with this Vector's extent.
+        :param geometry_type: Type of geometry to use for the proximity, defaults to 'boundary'.
+        :param in_or_out: Compute proximity only 'in' or 'out'-side the polygon, or 'both'.
+        :param distance_unit: Distance unit, either 'georeferenced' or 'pixel'.
+
+        :return: Proximity raster.
+        """
+
+        # TODO: (specific to Vector proximity) raster size could passed as argument when
+        #  nothing else is passed (here and in create_mask/rasterize)
+
+        # 0/ If no Raster is passed, create one on the Vector bounds of size 1000 x 1000
+        if raster is None:
+
+            # TODO: this bit of code is common in several vector functions (rasterize, etc): move out as common code?
+            # By default, use self's bounds
+            if self.bounds is None:
+                raise ValueError("To automatically rasterize on the vector, bounds need to be defined.")
+
+            # Calculate raster shape
+            left, bottom, right, top = self.bounds
+
+            # Calculate raster transform
+            transform = rio.transform.from_bounds(left, bottom, right, top, 1000, 1000)
+
+            raster = gu.Raster.from_array(data=np.zeros((1000, 1000)), transform=transform, crs=self.crs)
+
+        proximity = gu.georaster.raster.proximity_from_vector_or_raster(
+            raster=raster, vector=self, geometry_type=geometry_type,
+            in_or_out=in_or_out, distance_unit=distance_unit
+        )
+
+        return raster.copy(new_array=proximity)
+
+    def buffer_metric(self, buffer_size: float) -> Vector:
+        """
+        Buffer the vector in a metric system (UTM only).
+
+        The outlines are projected to a local UTM, then reverted to the original projection after buffering.
+
+        :param buffer_size: Buffering distance in meters
+
+        :return: Buffered shapefile
+        """
+
+        from geoutils.projtools import latlon_to_utm, utm_to_epsg
+
+        # Get a rough centroid in geographic coordinates (ignore the warning that it is not the most precise):
+        with warnings.catch_warnings():
+            warnings.simplefilter(action="ignore", category=UserWarning)
+            shp_wgs84 = self.ds.to_crs(epsg=4326)
+            lat, lon = shp_wgs84.centroid.y.values[0], shp_wgs84.centroid.x.values[0]
+
+        # Get the EPSG code of the local UTM
+        utm = latlon_to_utm(lat, lon)
+        epsg = utm_to_epsg(utm)
+
+        # Reproject the shapefile in the local UTM
+        ds_utm = self.ds.to_crs(epsg=epsg)
```

Comment:
same here, maybe use the syntax `with` to close it after the following line?

Reply:
Done too